### DISCUSSION TITLE
Pygments syntax highlighting for code blocks

### DISCUSSION THREAD

**Discussion by Qrbaker** [TrueBrain/TrueWiki#293] [Q&A] [answered]
As the Crafty team is setting up [our wiki](https://wiki.craftycontrol.com) some editors have requested Syntax highlighting.

I'm going to look into implementing this with [Pygments](https://pygments.org/docs/). This may be a major change and might add significant overhead to the server.

to that end:
1. Would you be interested in having syntax highlighting in upstream?
1. Should this be a toggle, i.e. with a command-line flag?

**Comment by TrueBrain** [comment, marked as answer]
Honestly, I would think any syntax highlighter should be client-side javascript, which can optionally be loaded by the client. I am not sure why this should be done server-side?

(I do not know Pygments, so I have no clue how it implemented it :P)

**Reply by Qrbaker** [reply]
Its done by emitting html/css: https://pygments.org/docs/quickstart/

The reason is for static page generation. Pygments is what powers syntax highlighting in [Hugo](https://gohugo.io/content-management/syntax-highlighting/#readout), for example.

It might make more sense to include in [wikitexthtml](../../wikitexthtml), actually.

**Reply by Qrbaker** [reply]
But yeah, [highlight.js](https://highlightjs.org/usage/) would work, too. I could see either.

**Reply by TrueBrain** [reply]
I would think syntax highlighting is only useful during editing, not prerendered? But I might be missing the use-case here .. could you explain a bit how this would be used? As in, where do those editors expect syntax highlighting exactly?

**Reply by TrueBrain** [reply]
> But yeah, [highlight.js](https://highlightjs.org/usage/) would work, too. I could see either.

Wauw, that website hurt my eyes.

You could also consider a nice editor like https://microsoft.github.io/monaco-editor/ instead of the textarea.

**Reply by Qrbaker** [reply]
I mean syntax editing for displaying code on the page itself, not in editing, actually. Some pages we are writing include code snippets. For an example, see this wikipedia page: https://en.wikipedia.org/wiki/Monad_(functional_programming)#An_example:_Maybe

**Reply by TrueBrain** [reply]
Ah, see, that is another use-case then I expected :D

Yeah, that belongs in wikitexthtml :)
